src/modules/accounting/prediction.py

```diff
@@ -0,0 +1,94 @@
+        """
+        :param contract_event: ETHDistributed(
+                uint256 indexed reportTimestamp,
+                uint256 preCLBalance,
+                uint256 postCLBalance,
+                uint256 withdrawalsWithdrawn
+                uint256 executionLayerRewardsWithdrawn
+                uint256 postBufferedEther
+            );
+        """
+        self.contract_event = contract_event
+
+    def get_rewards_per_epoch(self, report_timestamp: BlockStamp) -> Wei | Exception:
+        """
+            1. Read previous N report events.
+            2. Get median income
+            3. return median income / epoch_per_day
+
+
+
+            def fetch_predicted_pessimistic_rewards_per_epoch():
+              # Calcs events interval: ref_block - X days duration seconds / 12 ... ref_block
+              # Fetches `ETHDistributed` events from Lido contract
+              # Filters events by timestamp, since there could be missed blocks in slots and we may have captured more blocks
+              pass
+
+            def fetch_predicted_pessimistic_rewards_per_epoch(ref_block):
+              period = fetch_rewards_prediction_period(ref_block) // из контракта
+
+              # Rough estimate, since there could be missing blocks in the slots
+              max_period_block_range = period / SECONDS_PER_SLOT
+
+              from_block = ref_block - max_period_block_range
+              to_block = ref_block
+              events = fetch_eth_distributed_events(from_block, to_block)
+
+              # Narrow the events range
+              events = filter_events_by_timestamp(events)
+
+              return calc_average_rewards_per_epoch_from_events(events)
+        """
+
+        ## TODO: предел CL node на интервал блоков
+        from_block, to_block = self.get_block_for_filtering(report_timestamp['block_number'])
+
+        logs: Iterable[EventData]
+        try:
+            logs = self.contract_event.build_filter().get_logs(
+                from_block=from_block,
+                to_block=to_block,
+            )
+        except Exception as e:
+            return e
+
+        rewards: List[Wei] = list()
+        for log in logs:
+            reward = log['args']['withdrawalsWithdrawn'] + log['args']['executionLayerRewardsWithdrawn']
+            rewards.append(Wei(reward))
+
+        # Todo: ask analitics about how to calc
+        # Todo: different scenario about result
+        median: Wei = self.median(rewards)
+        return Wei(median)
+
+    @staticmethod
+    def get_block_for_filtering(ref_block: int) -> (int, int):
+        # Rough estimate, since there could be missing blocks in the slots
+        # 604800 / 12 = 50400
+        max_period_block_range = REWARDS_HISTORICAL_PERIOD_IN_SECONDS / SECONDS_PER_SLOT
+
+        from_block = ref_block - max_period_block_range
+        to_block = ref_block
+        return int(from_block), to_block
+
+    @staticmethod
+    def median(rewards: List[Wei]) -> Wei:
```

Comment:
probably it would be suitable
```
import statistics
statistics.median()
```

Reply:
@F4ever told that it's ok to use a small custom median function(cause it's pretty simple) instead of using additional lib.
Ps: the first variant was with using the external method from the library. Honestly, it was provided by "numpy".

Reply:
If this built-in lib why not? Numpy is not included in base python packages.
Let's write less code and use more available bicycles.

Reply:
And if you use lib, you don't have to test it

Reply:
yea, statistics is a built-in lib

Reply:
done